lib/src/components/supa_socials_auth.dart

```diff
@@ -147,75 +159,99 @@ class _SupaSocialsAuthState extends State<SupaSocialsAuth> {
+              redirectTo: widget.redirectUrl,
             );
-
-            foregroundColor = Colors.black;
-            backgroundColor = Colors.white;
-            overlayColor = Colors.white;
+          } on AuthException catch (error) {
+            if (widget.onError == null) {
+              context.showErrorSnackBar(error.message);
+            } else {
+              widget.onError?.call(error);
+            }
+          } catch (error) {
+            if (widget.onError == null) {
+              context
+                  .showErrorSnackBar('Unexpected error has occurred: $error');
+            } else {
+              widget.onError?.call(error);
+            }
           }
+        }
 
-          return Padding(
-            padding: const EdgeInsets.symmetric(vertical: 8.0),
-            child: ElevatedButton.icon(
-              icon: iconWidget,
-              style: ButtonStyle(
-                foregroundColor: MaterialStateProperty.all(foregroundColor),
-                backgroundColor: MaterialStateProperty.all(backgroundColor),
-                overlayColor: MaterialStatePropertyAll(overlayColor),
-                iconColor: MaterialStatePropertyAll(iconColor),
-              ),
-              onPressed: () async {
-                try {
-                  await supaClient.auth.signInWithOAuth(
-                    socialProvider.provider,
-                    redirectTo: widget.redirectUrl,
-                  );
-                } on AuthException catch (error) {
-                  if (widget.onError == null) {
-                    context.showErrorSnackBar(error.message);
-                  } else {
-                    widget.onError?.call(error);
-                  }
-                } catch (error) {
-                  if (widget.onError == null) {
-                    context.showErrorSnackBar(
-                        'Unexpected error has occurred: $error');
-                  } else {
-                    widget.onError?.call(error);
-                  }
-                }
-              },
-              label: Text('Continue with ${socialProvider.capitalizedName}'),
-            ),
-          );
-        },
-      ),
+        final authButtonStyle = ButtonStyle(
+          foregroundColor: MaterialStateProperty.all(foregroundColor),
+          backgroundColor: MaterialStateProperty.all(backgroundColor),
+          overlayColor: MaterialStateProperty.all(overlayColor),
+          iconColor: MaterialStateProperty.all(iconColor),
+        );
+
+        return Padding(
+          padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 4),
+          child: widget.socialButtonVariant == SocialButtonVariant.icon
+              ? ClipOval(
+                  child: Material(
+                    color: backgroundColor,
+                    child: InkWell(
+                      onTap: onAuthButtonPressed,
+                      child: iconWidget,
+                    ),
+                  ),
+                )
```

Comment:
I hope it was okay to do this, but I have updated the code to use Material and Inkwell to create a circle button. I always do this when I want to create a round button in Flutter, because this way the ripple does show up on top of the coloring and the icon.

![Mar-08-2023 14-57-07](https://user-images.githubusercontent.com/18113850/223631386-01763ab9-5b87-48f5-afa7-48cf73519ace.gif)

Reply:
no worries, at the end this is your repo. That wrapping with clipOal, material and Inkwell is a cool trick tho